AIDevGallery/Controls/SampleContainer.xaml.cs

```diff
@@ -397,26 +410,31 @@ private async void NuGetPackage_Click(object sender, RoutedEventArgs e)
         }
     }
 
-    private async void WindowsUpdateHyperlinkClicked(Microsoft.UI.Xaml.Documents.Hyperlink sender, Microsoft.UI.Xaml.Documents.HyperlinkClickEventArgs args)
+    private Task ReloadSampleAsync()
     {
-        var uri = new Uri("ms-settings:windowsupdate");
-        await Launcher.LaunchUriAsync(uri);
+        var models = _modelsCache;
+        var sample = _sampleCache;
+        _modelsCache = null;
+        _sampleCache = null;
+
+        return LoadSampleAsync(sample, models);
     }
 
     private async void WcrModelDownloader_DownloadClicked(object sender, EventArgs e)
     {
-        if (!LanguageModel.IsAvailable())
+        if (WcrApiHelpers.GetApiAvailability(_wcrApi) != WcrApiAvailability.Available)
         {
-            var op = LanguageModel.MakeAvailableAsync();
+            var op = WcrApiHelpers.MakeAvailables[_wcrApi]();
             if (await modelDownloader.SetDownloadOperation(op))
             {
                 // reload sample
-                _ = this.LoadSampleAsync(_sampleCache, _modelsCache);
+                _ = ReloadSampleAsync();
```

Comment:
Its better to await this so any exception is not swollen.

Reply:
Done